### PR TITLE
Reduce ERC20 allowance before triggering transfer 

### PR DESCRIPTION
**Pull request by frangio**
This has been pointed out many times and even though it is not an issue, I think it's better to change the order to follow the general rule of failing early.

### CODE REVIEW

File: test/token/ERC20/ERC20.behavior.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ function shouldBehaveLikeERC20 (errorPrefix, initialSupply, initialHolder, recip
       describe('when the recipient is not the zero address', function () {
         const to = anotherAccount;
 
-        describe('when the spender has enough approved balance', function () {
+        describe('when the spender has enough approved allowance', function () {
```
Comment (Amxx) [inline test/token/ERC20/ERC20.behavior.js:43]: "approved allowance" doesn't sound right to me. What about "when the spender has enough allowance"?

File: test/token/ERC20/ERC20.behavior.js [hunk 2]
```diff
@@ -84,37 +84,50 @@ function shouldBehaveLikeERC20 (errorPrefix, initialSupply, initialHolder, recip
...
+              await expectRevert(
+                this.token.transferFrom(tokenOwner, to, amount, { from: spender }),
+                `${errorPrefix}: transfer amount exceeds balance`,
               );
             });
           });
         });
 
-        describe('when the spender does not have enough approved balance', function () {
+        describe('when the spender does not have enough approved allowance', function () {
```
Comment (Amxx) [inline test/token/ERC20/ERC20.behavior.js:102]: same a above.

File: test/token/ERC20/ERC20.behavior.js [hunk 3]
```diff
@@ -84,37 +84,50 @@ function shouldBehaveLikeERC20 (errorPrefix, initialSupply, initialHolder, recip
...
               );
             });
           });
 
           describe('when the token owner does not have enough balance', function () {
-            const amount = initialSupply.addn(1);
+            const amount = allowance;
+
+            beforeEach('reducing balance', async function () {
+              await this.token.transfer(to, initialSupply, { from: tokenOwner });
```
Comment (Amxx) [inline test/token/ERC20/ERC20.behavior.js:124]: The other "reducing balance" `beforeEach` does a transfer of 1. This one does initialSupply. Can we make them more consistent ?

Comment (frangio) [reply test/token/ERC20/ERC20.behavior.js:124]: Can't make it 1 because the test passes, but I can make it 2.

**Comment by frangio** [comment]
@Amxx Your commits look good to me. Should we merge?

**Review by Amxx — APPROVED** [review]

**Pull request merged**